# HG changeset patch
# User HackaLittleBit
# Date 1516208362 0
#      Wed Jan 17 16:59:22 2018 +0000
# Branch trunk
# Node ID c00a3190d8333123d202032eefa7ae630ff72520
# Parent  c2b348bc2c2f09e01b3db5cd9542f15686fd3a63
Add: Give warning when requested custom town quantity was not met.

diff -r c2b348bc2c2f -r c00a3190d833 src/lang/english.txt
--- a/src/lang/english.txt	Tue Jan 16 14:23:46 2018 +0000
+++ b/src/lang/english.txt	Wed Jan 17 16:59:22 2018 +0000
@@ -4096,6 +4096,7 @@
 
 # Map generation messages
 STR_ERROR_COULD_NOT_CREATE_TOWN                                 :{WHITE}Map generation aborted...{}... no suitable town locations
+STR_ERROR_COULD_NOT_CREATE_TOWNS_ENOUGH                         :{WHITE}Only able to create {NUM} town(s). Change the map generation parameters to get a better map
 STR_ERROR_NO_TOWN_IN_SCENARIO                                   :{WHITE}... there is no town in this scenario
 
 STR_ERROR_PNGMAP                                                :{WHITE}Can't load landscape from PNG...
diff -r c2b348bc2c2f -r c00a3190d833 src/town_cmd.cpp
--- a/src/town_cmd.cpp	Tue Jan 16 14:23:46 2018 +0000
+++ b/src/town_cmd.cpp	Wed Jan 17 16:59:22 2018 +0000
@@ -1971,6 +1971,13 @@
 
 	town_names.clear();
 
+	if (current_number < _settings_newgame.game_creation.custom_town_number &&
+			difficulty == (uint)CUSTOM_TOWN_NUMBER_DIFFICULTY &&
+			_game_mode != GM_EDITOR) {
+		SetDParam(0, current_number);
+		ShowErrorMessage(STR_ERROR_COULD_NOT_CREATE_TOWNS_ENOUGH, INVALID_STRING_ID, WL_WARNING);
+	}
+
 	if (current_number != 0) return true;
 
 	/* If current_number is still zero at this point, it means that not a single town has been created.
